omp/matrix/dense_kernels.cpp

```diff
@@ -472,8 +472,11 @@ void convert_to_sellp(std::shared_ptr<const OmpExecutor> exec,
             }
         }
     }
-    slice_sets[slice_num] =
-        slice_sets[slice_num - 1] + slice_lengths[slice_num - 1];
+
+    if (slice_num > 0) {
+        slice_sets[slice_num] =
+            slice_sets[slice_num - 1] + slice_lengths[slice_num - 1];
+    }
```

Comment:
I think it would be nice if only for the conversion to sellp you could check at least for either CUDA or HIP that empty conversion also works, since you had to modify the kernels here for this. I looked only quickly and it should be fine as long as num_rows = 0, but making sure of it would maybe be a good idea.